### PR TITLE
feat: add DesignSystemResolver formatting utility

### PR DESCRIPTION
**Pull request by nicholasrice**
# Description
The formatter allows us to format an input string with designSystemResolvers, meaning we don't need access to a DesignSystem value to format properties like border. This will help enable us to move away from function stylesheets, which are less performant than object stylesheets when design-system values change at runtime.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-styles-msft/src/text-action/index.ts [hunk 1]
```diff
@@ -60,9 +62,14 @@ const styles: ComponentStyles<TextActionClassNameContract, DesignSystem> = (
...
-                designSystem
-            )}`,
+            border: format(
+                "{0} solid {1}",
+                toPx<DesignSystem>(outlineWidth),
+                neutralOutlineRest
+            ),
+            // border: `${toPx(designSystem.outlineWidth)} solid ${neutralOutlineRest(
+            //     designSystem
+            // )}`,
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/text-action/index.ts]: still needed?

File: packages/fast-components-styles-msft/src/utilities/design-system.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                return designSystem.outlineWidth * arg;
+            }
+        );
+    }
+
+    return withDesignSystemDefaults(arg).outlineWidth;
+}
+
+/**
+ * Retrieve the outlineWidth when invoked with a DesignSystem
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/utilities/design-system.ts]: outlineWidth? Think you might mean designUnit here

Comment (chrisdholt) [inline packages/fast-components-styles-msft/src/utilities/design-system.ts]: Can we update this to reflect the function below?

File: packages/fast-components-styles-msft/src/utilities/format.spec.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            formatter(designSystemDefaults) + formatter(designSystemDefaults)
+        );
+    });
+    test("should not format a variable when not supplied a formatter", (): void => {
+        expect(format("{0}{1}", formatter)(designSystemDefaults)).toBe(
+            formatter(designSystemDefaults) + "{1}"
+        );
+    });
+    test("should format multiple variables when multiple formatters are passed", (): void => {
+        expect(format("{0}{1}", formatter, formatter)(designSystemDefaults)).toBe(
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/utilities/format.spec.ts]: I would suggest using a different formatter for the second formatter, just to make sure there isn't an issue where the wrong formatter gets used

File: packages/fast-components-styles-msft/src/text-action/index.ts [hunk 4]
```diff
@@ -60,9 +62,14 @@ const styles: ComponentStyles<TextActionClassNameContract, DesignSystem> = (
...
             minWidth: "92px",
-            border: `${toPx(designSystem.outlineWidth)} solid ${neutralOutlineRest(
-                designSystem
-            )}`,
+            border: format(
+                "{0} solid {1}",
+                toPx<DesignSystem>(outlineWidth),
+                neutralOutlineRest
+            ),
+            // border: `${toPx(designSystem.outlineWidth)} solid ${neutralOutlineRest(
```
Comment (chrisdholt) [inline packages/fast-components-styles-msft/src/text-action/index.ts]: I think we can remove this

File: packages/fast-components-styles-msft/src/context-menu-item/index.ts [hunk 5]
```diff
@@ -11,63 +15,64 @@ import {
...
+            color: neutralForegroundActive,
+            background: neutralFillStealthActive,
         },
-    };
+    },
+    contextMenuItem_contentRegion: {
+        gridColumnStart: "2",
+        justifySelf: "start",
+        overflow: "hidden",
+        textOverflow: "ellipsis",
```
[the remark this answers is not in the document]
Comment (nicholasrice) [reply packages/fast-components-styles-msft/src/context-menu-item/index.ts]: this is a copy+pasta so I'll leave this as-is

**Pull request merged**